actor/src/main/scala/org/apache/pekko/pattern/RetrySupport.scala

```diff
@@ -148,21 +149,23 @@ trait RetrySupport {
       implicit
       ec: ExecutionContext,
       scheduler: Scheduler): Future[T] = {
-    RetrySupport.retry(attempt, attempts, delayFunction, attempted = 0)
+    RetrySupport.retry(attempt, attempts, delayFunction, attempted = 0, RetrySupport.retryOnException)
   }
 }
 
 object RetrySupport extends RetrySupport {
+  private val retryOnException: (Any, Throwable) => Boolean = (_: Any, e: Throwable) => e != null
 
   private def retry[T](attempt: () => Future[T], maxAttempts: Int, attempted: Int)(
       implicit ec: ExecutionContext): Future[T] =
-    retry(attempt, maxAttempts, ConstantFun.scalaAnyToNone, attempted)(ec, null)
+    retry(attempt, maxAttempts, ConstantFun.scalaAnyToNone, attempted, retryOnException)(ec, null)
 
   private def retry[T](
       attempt: () => Future[T],
       maxAttempts: Int,
       delayFunction: Int => Option[FiniteDuration],
-      attempted: Int)(implicit ec: ExecutionContext, scheduler: Scheduler): Future[T] = {
+      attempted: Int,
+      shouldRetry: (T, Throwable) => Boolean)(implicit ec: ExecutionContext, scheduler: Scheduler): Future[T] = {
```

Comment:
I think a `(T,Throwable) =>Boolean` will be more helpful than `Throwable =>Boolean`.